mozilla_django_oidc_db/config.py

```diff
@@ -110,51 +124,53 @@ def store_config(request: HttpRequest) -> None:
     mozilla-django-oidc's callback view deletes the state key after it has validated it,
     so our :func:`lookup_config` cannot extract it from the session anymore.
     """
-    # Attempt to retrieve the config_class from the session, this only works for users
+    # Attempt to retrieve the config_identifier from the session, this only works for users
     # that are actually logged in as Django users
-    # The config_class key is added to the state in the OIDCInit.get method.
-    # TODO: verify that the state query param is present for error flows! Need to check
-    # the OAUTH2 spec for this, but according to ChatGeePeeTee if the request contains
-    # it, the callback must have it too.
-    config_class = ""
+    # The config_identifier key is added to the state in the OIDCAuthenticationRequestInitView.get method.
+    # The state parameter is present in the error flow if it was present in the Authorization request: https://www.rfc-editor.org/rfc/rfc6749#section-4.1.2.1
+    config_identifier = ""
     state_key = request.GET.get("state")
     if state_key and state_key in (states := request.session.get("oidc_states", [])):
         state = states[state_key]
-        config_class = state.get("config_class", "")
+        config_identifier = state.get("config_identifier", "")
 
-    if not config_class and (
-        _config := request.session.get(CONFIG_CLASS_SESSION_KEY, "")
+    if not config_identifier and (
+        _config := request.session.get(CONFIG_IDENTIFIER_SESSION_KEY, "")
     ):
-        config_class = _config
+        config_identifier = _config
 
     try:
-        config = apps.get_model(config_class)
-    except (LookupError, ValueError) as exc:
-        raise BadRequest("Could not look up the referenced config.") from exc
+        config = OIDCConfig.objects.select_related("oidc_provider_config").get(
+            identifier=config_identifier
+        )
+    except (ObjectDoesNotExist, MultipleObjectsReturned) as exc:
```

Comment:
```suggestion
    except (OIDCConfig.DoesNotExist, MultipleObjectsReturned) as exc:
```
you always want to favour the model-specific exception instead of the base class for it.